flutter_local_notifications_linux/lib/src/storage.dart

```diff
@@ -0,0 +1,178 @@
+    FileSystem? fs,
+  })  : _platformInfo = platformInfo ?? LinuxPlatformInfo(),
+        _fs = fs ?? LocalFileSystem();
+
+  final LinuxPlatformInfo _platformInfo;
+  final FileSystem _fs;
+
+  _Cache? _cachedInfo;
+
+  /// Get all notifications.
+  Future<List<LinuxNotificationInfo>> getAll() async {
+    final _Cache cache = await _readInfoMap();
+    return cache.toImmutableMap().values.toList();
+  }
+
+  /// Get notification by [LinuxNotificationInfo.id].
+  Future<LinuxNotificationInfo?> getBySystemId(int systemId) async {
+    final _Cache cache = await _readInfoMap();
+    return cache.getBySystemId(systemId);
+  }
+
+  /// Get notification by [LinuxNotificationInfo.systemId].
+  Future<LinuxNotificationInfo?> getById(int id) async {
+    final _Cache cache = await _readInfoMap();
+    return cache.getById(id);
+  }
+
+  /// Insert notification to the storage.
+  /// Returns `true` if the operation succeeded.
+  Future<bool> insert(LinuxNotificationInfo notification) async {
+    final _Cache cache = await _readInfoMap();
+    cache.insert(notification);
+    return _writeInfoList(cache.values.toList());
+  }
+
+  /// Remove notification from the storage by [LinuxNotificationInfo.id].
+  /// Returns `true` if the operation succeeded.
+  Future<bool> removeById(int id) async {
+    final _Cache cache = await _readInfoMap();
+    cache.removeById(id);
+    return _writeInfoList(cache.values.toList());
+  }
+
+  /// Remove notification from the storage by [LinuxNotificationInfo.systemId].
+  /// Returns `true` if the operation succeeded.
+  Future<bool> removeBySystemId(int systemId) async {
+    final _Cache cache = await _readInfoMap();
+    final LinuxNotificationInfo? info = cache.getBySystemId(systemId);
+    if (info != null) {
+      cache.removeById(info.id);
+    }
+    return _writeInfoList(cache.values.toList());
+  }
+
+  /// Remove notification from the storage by [idList].
+  /// Returns `true` if the operation succeeded.
+  Future<bool> removeByIdList(List<int> idList) async {
+    final _Cache cache = await _readInfoMap();
+    // ignore: prefer_foreach
+    for (final int id in idList) {
+      cache.removeById(id);
+    }
+    return _writeInfoList(cache.values.toList());
+  }
+
+  /// Force read info from the disk to the cache.
+  Future<void> forceReloadCache() async {
+    _cachedInfo = await _readToCache();
+  }
+
+  Future<File?> _getStorageFile() async {
+    final LinuxPlatformInfoData data = await _platformInfo.getAll();
+    final String? dir = data.runtimePath;
+    if (dir == null) {
+      return null;
+    }
+    return _fs.open(path.join(dir, _kFileName));
+  }
+
+  /// Gets a [LinuxNotificationInfo] from the stored file.
+  /// Once read, the data are maintained in memory.
+  Future<_Cache> _readInfoMap() async {
+    if (_cachedInfo != null) {
+      return _cachedInfo!;
+    }
+    return _cachedInfo = await _readToCache();
+  }
+
+  Future<_Cache> _readToCache() async {
```

Comment:
aorry missed this before but in terms of grammar this should be `_readFromCache`

Reply:
Yeah, sorry)